consensus/vbft/service.go

```diff
@@ -1783,8 +1783,15 @@ func (self *Server) processTimerEvent(evt *TimerEvent) error {
 	case EventTxPool:
 		self.timer.stopTxTicker(evt.blockNum)
 		if self.completedBlockNum+1 == evt.blockNum {
-			txpool := self.poolActor.GetTxnPool(true, self.validHeight(evt.blockNum))
-			if len(txpool) != 0 {
+			validHeight := self.validHeight(evt.blockNum)
+			newProposal := false
+			for _, e := range self.poolActor.GetTxnPool(true, uint32(validHeight)) {
```

Comment:
validHeight has been uint32.  
type cast not needed?

Reply:
yes